opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -0,0 +1,299 @@
+      FormatNames.BASIC_TIME_NO_MILLIS,
+      FormatNames.BASIC_T_TIME,
+      FormatNames.BASIC_T_TIME_NO_MILLIS,
+      FormatNames.TIME,
+      FormatNames.STRICT_TIME,
+      FormatNames.TIME_NO_MILLIS,
+      FormatNames.STRICT_TIME_NO_MILLIS,
+      FormatNames.HOUR_MINUTE_SECOND_FRACTION,
+      FormatNames.STRICT_HOUR_MINUTE_SECOND_FRACTION,
+      FormatNames.HOUR_MINUTE_SECOND_MILLIS,
+      FormatNames.STRICT_HOUR_MINUTE_SECOND_MILLIS,
+      FormatNames.HOUR_MINUTE_SECOND,
+      FormatNames.STRICT_HOUR_MINUTE_SECOND,
+      FormatNames.HOUR_MINUTE,
+      FormatNames.STRICT_HOUR_MINUTE,
+      FormatNames.HOUR,
+      FormatNames.STRICT_HOUR,
+      FormatNames.T_TIME,
+      FormatNames.STRICT_T_TIME,
+      FormatNames.T_TIME_NO_MILLIS,
+      FormatNames.STRICT_T_TIME_NO_MILLIS
+  );
+
+  @EqualsAndHashCode.Exclude
+  String formatString;
+
+  private OpenSearchDateType() {
+    super(MappingType.Date);
+    this.formatString = "";
+  }
+
+  private OpenSearchDateType(ExprCoreType exprCoreType) {
+    super(MappingType.Date);
+    this.formatString = "";
+    this.exprCoreType = exprCoreType;
+  }
+
+  private OpenSearchDateType(ExprType exprType) {
+    super(MappingType.Date);
+    this.formatString = "";
+    this.exprCoreType = (ExprCoreType) exprType;
+  }
+
+  private OpenSearchDateType(String formatStringArg) {
+    super(MappingType.Date);
+    this.formatString = formatStringArg;
+    this.exprCoreType = getExprTypeFromFormatString(formatStringArg);
+  }
+
+  /**
+   * Retrieves and splits a user defined format string from the mapping into a list of formats.
+   * @return A list of format names and user defined formats.
+   */
+  private List<String> getFormatList() {
+    String format = strip8Prefix(formatString);
+    List<String> patterns = splitCombinedPatterns(format);
+    return patterns;
+  }
+
+
+  /**
+   * Retrieves a list of named formatters defined by OpenSearch.
+   * @return a list of DateFormatters that can be used to parse a Date/Time/Timestamp.
+   */
+  public List<DateFormatter> getAllNamedFormatters() {
+    return getFormatList().stream()
+        .filter(formatString -> FormatNames.forName(formatString) != null)
+        .map(DateFormatter::forPattern).collect(Collectors.toList());
+  }
+
+  /**
+   * Retrieves a list of custom formatters defined by the user.
+   * @return a list of DateFormatters that can be used to parse a Date/Time/Timestamp.
+   */
+  public List<DateFormatter> getAllCustomFormatters() {
+    return getFormatList().stream()
+        .filter(formatString -> FormatNames.forName(formatString) == null)
+        .map(DateFormatter::forPattern).collect(Collectors.toList());
+  }
+
+  /**
+   * Retrieves a list of named formatters that format for dates.
+   *
+   * @return a list of DateFormatters that can be used to parse a Date.
+   */
+  public List<DateFormatter> getDateNamedFormatters() {
+    return getFormatList().stream()
+        .filter(formatString -> {
```

Comment:
It is good to filter `getAllNamedFormatters` here

Reply:
Will be included in a follow up PR including user defined format